Makefile

```diff
@@ -236,8 +236,9 @@ depend: init | $(ISTIO_OUT)
 $(ISTIO_OUT) $(ISTIO_BIN):
 	@mkdir -p $@
 
-$(ISTIO_OUT)/dep.png: $(ISTIO_OUT)/dep.dot
-	dot -T png < $(ISTIO_OUT)/dep.dot > $(ISTIO_OUT)/dep.png
+# Updates the dependencies and generates a git diff of the vendor files against HEAD.
+depend.diff: depend.update | $(ISTIO_OUT)
```

Comment:
We don't want each PR to do a depend.update. That's why we have vendor in the first place - so people don't have to do dep update every time.

Reply:
`depend.update` shouldn't do anything.  This ensures that doing `dep ensure` does not change `Gopkg.lock` or `vendor`.

Reply:
As discussed offline, this target is only run from nightly and periodic builds.

Reply:
Well - after our offline talk I thought about your arguments, and I think I was wrong...

How about: add this as a dependency to 'lint', which is already run and a requirement. It is a form of linting - and won't have a big impact. 

My concern was more with having it as a separate job/container and in front of other jobs - but as part of lint it seems quite reasonable.

Reply:
So you're no longer concerned with running `dep ensure` for every PR?

@rshriram WDYT?

Reply:
I think there is quite a bit of confusion here, and I don't see how this test can even work.

'depend.update' is supposed to pull very last version of all dependencies. It just can't match what the developer did - many of the deps might have had pushes. So doing a diff may work in rare cases, but will fail most of the time. 

Also the purpose of depend.update is to verify recent versions of deps don't break us.

The diff is doing a very different thing - verifies that the lock and vendor are matching. I am ok with doing this diff as part of the lint job.